internal/services/automation/automation_source_control.go

```diff
@@ -0,0 +1,299 @@
+package automation
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/automation/mgmt/2020-01-13-preview/automation"
+
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/automation/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/automation/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type SecurityToken struct {
+	Token        string `tfschema:"token"`
+	RefreshToken string `tfschema:"refresh_token"`
+	TokenType    string `tfschema:"token_type"`
+}
+
+type SourceControlModel struct {
+	ResourceGroupName     string          `tfschema:"resource_group_name"`
+	AutomationAccountName string          `tfschema:"automation_account_name"`
+	Name                  string          `tfschema:"name"`
+	RepoURL               string          `tfschema:"repo_url"`
+	Branch                string          `tfschema:"branch"`
+	FolderPath            string          `tfschema:"folder_path"`
+	AutoSync              bool            `tfschema:"auto_sync"`
+	PublishRunbook        bool            `tfschema:"publish_runbook"`
+	SourceType            string          `tfschema:"source_type"`
+	Description           string          `tfschema:"description"`
+	SecurityToken         []SecurityToken `tfschema:"security_token"`
+}
+
+type SourceControlResource struct{}
+
+var _ sdk.Resource = (*SourceControlResource)(nil)
+
+func (m SourceControlResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"resource_group_name": commonschema.ResourceGroupName(),
+		"automation_account_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.AutomationAccount(),
+		},
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotWhiteSpace,
+		},
+		"repo_url": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringLenBetween(0, 2000),
+		},
+		"branch": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringLenBetween(0, 255),
+		},
+		"folder_path": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringLenBetween(0, 255),
+		},
+		"auto_sync": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  false,
+		},
+		"publish_runbook": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  true,
+		},
+		"source_type": {
```

Comment:
```suggestion
		"source_control_type": {
```

this is the vcs type?